### DISCUSSION TITLE
Question regards to train/val/test split

### DISCUSSION THREAD

**Discussion by seanko29** [open-edge-platform/anomalib#761] [Q&A] [answered]
Hello. I have a question (maybe very newbie one T^T) regards to train/val/test split.
I read the question & answers in #539 and have extra questions on this.
 
The image below is my custom dataset. 
So, I have a normal image folder which I ONLY want to use for training purpose.
I have a test normal/abnormal directory separately constructed.
If I put the `split_ratio `into `null `and specify `abnormal_dir `and `normal_test_dir` separately and make `create_validation_set `as **true**, will the library automatically get 50% of validation from `abnormal_dir `and `normal_test_dir`??

I do not want the model to get the split from the train data into the validation or test set. 

So, in summary, train dataset for train dataset only and other abnormal / test dir for both val&test only.
Thank you in advanced :)

![image](https://user-images.githubusercontent.com/72428646/205627777-c47683a7-1b1d-4242-be52-05b1a4a7ca9d.png)

**Comment by djdameln** [comment, marked as answer]
> If I put the split_ratio into null and specify abnormal_dir and normal_test_dir separately and make create_validation_set as true, will the library automatically get 50% of validation from abnormal_dir and normal_test_dir??

Yes, when `create_validation_set` is set to `true`, the evaluation images will be divided equally (50%) into a validation set and a test set which are mutually exclusive. When set to `false`, the test set and the validation set will both contain the same images.

> I do not want the model to get the split from the train data into the validation or test set.

The settings you mentioned should achieve this. Whenever a `normal_test_dir` is specified, the normal images for evaluation (validation+testing) will all be taken from the `normal_test_dir`, and all normal images in `normal_dir` will be reserved exclusively for the training set. You do not even need to set `split_ratio` to `null` in this case, the parameter will just be ignored.

Please note that from v0.4.0, the naming of some of these parameters will change. `create_validation_set` will be deprecated in favor of `validation_split_mode`, and `split_ratio` will be renamed to `normal_split_ratio`. In addition, a `val_split_ratio` parameter will be added to change the val/test split to something other than 50%. Some documentation on this new configuration convention will follow.

**Reply by seanko29** [reply]
Thank you for the comment @djdameln !! Really appreciate it!
I see that the current version is v0.3.7. Will you upgrade it straight to v0.4.0 ?

Looking forward to the new version as well!! :D

**Reply by seanko29** [reply]
@djdameln  Just a follow-up question, if you do not mind.
IF I leave the path as  ' ' (empty string) like above and just directly mention `normal_dir`, will it automatically find the training path?

**Reply by djdameln** [reply]
Yes, that should work as well. If the `path` parameter is empty, you may pass the absolute path to the train and test images.
